components/brave_rewards/browser/rewards_service_impl.h

```diff
@@ -362,8 +361,9 @@ class RewardsServiceImpl : public RewardsService,
 
   void OnPreferenceChanged(const std::string& key);
 
-  void CheckPreferences();
-
+  void StartProcess(base::OnceClosure callback);
```

Comment:
It took me a bit of thinking to figure out how all of the "start" methods interact, and I'm wondering if we can simplify things down further. I'm wondering if we can we get by with just doing:

- Remove existing `StartProcess` completely, and all current calls.
- In `CheckPreferences`, call `StartLedgerProcess` if `kEnabled` is true.
- In `CreateRewardsWallet`, inline the current `StartProcess` body.

Reply:
Updated in a65ccdd5748fb3b9a90a8933c54151b401792223